pkg/pgxconn/implement.go

```diff
@@ -0,0 +1,57 @@
+package pgxconn
+
+import (
+	"time"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+)
+
+// rowWithMetric wraps the row returned by QueryRow() for metrics telemetry.
+type rowWithMetric struct {
+	r pgx.Row
+}
+
+func (w rowWithMetric) Scan(dest ...interface{}) error {
+	err := w.r.Scan(dest...)
+	if err != nil && err != pgx.ErrNoRows {
+		errorsTotal.With(promMethodLabel("query_row")).Inc()
+	}
+	return err
+}
+
+// batchResultsWithDuration wraps the SendBatch and records the duration of the batch till it closed.
+type batchResultsWithDuration struct {
+	b     pgx.BatchResults
+	start time.Time
+}
+
+func newBatchResultsWithDuration(b pgx.BatchResults) batchResultsWithDuration {
+	return batchResultsWithDuration{
+		b:     b,
+		start: time.Now(),
+	}
+}
+
+func (w batchResultsWithDuration) Close() error {
+	defer func() {
+		duration.With(promMethodLabel("send_batch"))
```

Comment:
I am not sure what is the goal here? We set the label but don't record duration?
Semantics seems unclear as this clearly does not measure `sendBatch` but actual processing of batch results. Let's discuss this more offline.

Reply:
I missed that, thank you!